test/e2e/autonomy/autonomy_test.go

```diff
@@ -0,0 +1,330 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yurthub
+
+import (
+	"context"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	"github.com/openyurtio/openyurt/test/e2e/common/ns"
+	p "github.com/openyurtio/openyurt/test/e2e/common/pod"
+	"github.com/openyurtio/openyurt/test/e2e/util"
+	"github.com/openyurtio/openyurt/test/e2e/util/ginkgowrapper"
+	ycfg "github.com/openyurtio/openyurt/test/e2e/yurtconfig"
+)
+
+const (
+	YurtE2ENamespaceName     = "yurt-e2e-test"
+	YurtDefaultNamespaceName = "default"
+	YurtSystemNamespaceName  = "kube-system"
+	YurtCloudNodeName        = "openyurt-e2e-test-control-plane"
+	YurtEdgeNodeName         = "openyurt-e2e-test-worker"
+	YurtEdgeNode2Name        = "openyurt-e2e-test-worker2"
+	NginxServiceName         = "yurt-e2e-test-nginx"
+	CoreDNSServiceName       = "kube-dns"
+)
+
+var (
+	c                clientset.Interface
+	err              error
+	Edge2NginxPodIP  string
+	NginxServiceIP   string
+	CoreDNSServiceIP string
+
+	flannelContainerID   string
+	yurthubContainerID   string
+	kubeProxyContainerID string
+	coreDnsContainerID   string
+	coreDnsNodeName      string
+	nginxContainerID     string
+)
+
+var _ = ginkgo.Describe("edge-autonomy"+YurtE2ENamespaceName, ginkgo.Ordered, ginkgo.Label("edge-autonomy"), func() {
+	defer ginkgo.GinkgoRecover()
+	var _ = ginkgo.Describe("kubelet"+YurtE2ENamespaceName, func() {
+		gomega.RegisterFailHandler(ginkgowrapper.Fail)
+		ginkgo.It("kubelet edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
+			// restart kubelet using systemctl restart kubelet in edge nodes；check if kubelet restarted periodically
+			gomega.Eventually(func() bool {
+				_, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'systemctl restart kubelet'").CombinedOutput()
```

Comment:
This command should be execute only once, please use Expect instead.

Reply:
will do